File: dmn-core/src/test/resources/tck/cl3/expected/dmn/0004-lending/CreditContingencyFactorTableRuleOutput.java

import java.util.*;

@javax.annotation.Generated(value = {"decisionTableRuleOutput.ftl", "CreditContingencyFactorTable"})
public class CreditContingencyFactorTableRuleOutput extends com.gs.dmn.runtime.RuleOutput {
    private java.math.BigDecimal creditContingencyFactorTable;

    public CreditContingencyFactorTableRuleOutput(boolean matched) {
        super(matched);
    }

    public java.math.BigDecimal getCreditContingencyFactorTable() {
        return this.creditContingencyFactorTable;
    }
    public void setCreditContingencyFactorTable(java.math.BigDecimal creditContingencyFactorTable) {
        this.creditContingencyFactorTable = creditContingencyFactorTable;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        CreditContingencyFactorTableRuleOutput other = (CreditContingencyFactorTableRuleOutput) o;
        if (this.getCreditContingencyFactorTable() != null ? !this.getCreditContingencyFactorTable().equals(other.getCreditContingencyFactorTable()) : other.getCreditContingencyFactorTable() != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = 0;
        result = 31 * result + (this.getCreditContingencyFactorTable() != null ? this.getCreditContingencyFactorTable().hashCode() : 0);

        return result;
    }

    public String toString() {
        StringBuilder result_ = new StringBuilder("(matched=" + isMatched());
        result_.append(String.format(", creditContingencyFactorTable='%s'", creditContingencyFactorTable));
        result_.append(")");
        return result_.toString();
    }
}

File: dmn-test-cases/standard/proto/1.1/0004-lending/translator/expected/proto3/kotlin/dmn/CreditContingencyFactorTableRuleOutput.kt

@javax.annotation.Generated(value = ["decisionTableRuleOutput.ftl", "CreditContingencyFactorTable"])
class CreditContingencyFactorTableRuleOutput(matched: Boolean) : com.gs.dmn.runtime.RuleOutput(matched) {
    @com.fasterxml.jackson.annotation.JsonProperty("CreditContingencyFactorTable")
    var creditContingencyFactorTable: java.math.BigDecimal? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as CreditContingencyFactorTableRuleOutput
        if (if (this.creditContingencyFactorTable != null) this.creditContingencyFactorTable != other.creditContingencyFactorTable else other.creditContingencyFactorTable != null) return false

        return true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (if (this.creditContingencyFactorTable != null) this.creditContingencyFactorTable.hashCode() else 0)

        return result
    }

    override fun toString(): String {
        val result_ = StringBuilder("(matched=" + isMatched())
        result_.append(String.format(", creditContingencyFactorTable='%s'", creditContingencyFactorTable))
        result_.append(")")
        return result_.toString()
    }
}
